src/js/tech/tech.js

```diff
@@ -205,7 +242,40 @@ class Tech extends Component {
       if (bufferedPercent === 1) {
         this.stopTrackingProgress();
       }
-    }, 500);
+    }), 500);
+  }
+
+  onDurationChange() {
+    duration_ = this.duration();
+  }
+
+  bufferedPercent() {
```

Comment:
If this is being duplicated from the player. If so we should probably create a utility for it so it can be shared.

Also it doesn't look like `duration_` is ever defined in this instance.